static/elements/chromedash-approvals-dialog.js

```diff
@@ -1,42 +1,368 @@
 import {LitElement, css, html} from 'lit-element';
+import {nothing} from 'lit-html';
 import './chromedash-dialog';
 import SHARED_STYLES from '../css/shared.css';
 
+const STATE_NAMES = [
+  [0, 'Needs review'],
+  [1, 'N/a or Ack'],
+  [3, 'Review started'],
+  [4, 'Need info'],
+  [5, 'Approved'],
+  [6, 'Not approved'],
+];
+const ACTIVE_STATES = [0, 3, 4];
+
+const APPROVAL_DEFS = [
+  {name: 'Intent to Prototype',
+    id: 1,
+    threadField: 'intent_to_implement_url',
+  },
+  {name: 'Intent to Experiment',
+    id: 2,
+    threadField: 'intent_to_experiment_url',
+  },
+  {name: 'Intent to Extend Experiment',
+    id: 3,
+    threadField: 'intent_to_extend_experiment_url',
+  },
+  {name: 'Intent to Ship',
+    id: 4,
+    threadField: 'intent_to_ship_url',
+  },
+];
+
+
 class ChromedashApprovalsDialog extends LitElement {
   static get properties() {
     return {
       signedInUser: {type: String},
       featureId: {type: Number},
       canApprove: {type: Boolean},
+      feature: {type: Object},
+      approvals: {type: Array},
+      comments: {type: Array},
+      showAllIntents: {type: Boolean},
+      changedApprovalsByField: {attribute: false},
+      needsSave: {type: Boolean},
     };
   }
 
   constructor() {
     super();
     this.signedInUser = ''; // email address
     this.canApprove = false;
+    this.featureId = 0;
+    this.feature = {};
+    this.approvals = [];
+    this.comments = [];
+    this.subsetPending = false;
+    this.showAllIntents = false;
+    this.changedApprovalsByField = new Map();
+    this.needsSave = false;
   }
 
   static get styles() {
     return [
       SHARED_STYLES,
       css`
+        h3 {
+          margin: var(--content-padding-half);
+        }
+
+        .approval_section {
+          margin-top: var(--content-padding);
+        }
+
+        .comment_section {
+          max-height: 250px;
+          overflow-y: scroll;
+        }
+
+        .approval_section div,
+        .comment {
+          margin-left: var(--content-padding);
+        }
+
+        .approval_row {
+          width: 650px;
+          margin-bottom: var(--content-padding-half);
+        }
+
+        .set_by,
+        .set_on,
+        .appr_val {
+          display: inline-block;
+          width: 200px;
+          margin-right: var(--content-padding-half);
+          overflow: hidden;
+          text-overflow: ellipsis;
+          white-space: nowrap;
+        }
+
+        select {
+          margin: 0;
+        }
+
+        .comment_body {
+          background: var(--table-alternate-background);
+          padding: var(--content-padding-half);
+          white-space: pre-wrap;
+          width: 46em;
+          margin-bottom: var(--content-padding);
+        }
+
+        .controls {
+          padding: var(--content-padding);
+          text-align: right;
+        }
+
+        #show_all_checkbox {
+         float: left;
+        }
+
+        textarea {
+          padding: 4px;
```

Comment:
Maybe add `resize: vertical or none`

Reply:
I went with `resize: both`. I figure, let the user have the power to do that if they want to.